apps/client/src/common/hooks/useEventAction.ts

```diff
@@ -152,6 +153,54 @@ export const useEventAction = () => {
     [_updateEventMutation],
   );
 
+  /**
+   * Calls mutation to edit multiple events
+   * @private
+   */
+  const _batchUpdateEventsMutation = useMutation({
+    mutationFn: requestBatchPutEvents,
+    onMutate: async ({ ids, data }) => {
+      // cancel ongoing queries
+      await queryClient.cancelQueries([RUNDOWN_TABLE_KEY]);
+
+      // Snapshot the previous value
+      const previousEvents = queryClient.getQueryData([RUNDOWN_TABLE_KEY]) as OntimeRundown;
+
+      const updatedEvents = previousEvents.map((event) => {
+        const isEventEdited = ids.includes(event.id);
+
+        if (isEventEdited && isOntimeEvent(event)) {
+          return {
+            ...event,
+            // spread new changed properties, overwriting old event properties
+            ...data,
+          };
+        }
+
+        return event;
+      });
+
+      queryClient.setQueryData(RUNDOWN_TABLE, updatedEvents);
+
+      // Return a context with the previous and new events
+      return { previousEvent: previousEvents };
+    },
+    onSettled: async () => {
+      await queryClient.invalidateQueries([RUNDOWN_TABLE_KEY]);
+    },
+  });
+
+  const batchUpdateEvents = useCallback(
+    async (data: Partial<OntimeRundownEntry>, eventIds: string[]) => {
+      try {
+        await _batchUpdateEventsMutation.mutateAsync({ ids: eventIds, data });
```

Comment:
quick thought:
- the user has a rundown of 10 elements
- they want to make the whole thing public, so they use multi-select to select the range and make it public
- it could be that a part of them were already public 

question:
Would it make sense to quickly check if the change is different from the initial value? either here, in backend, or in both places?
Some of these changes could cause chain calculations which would be good to avoid